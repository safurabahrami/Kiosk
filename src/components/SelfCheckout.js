import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';

import * as Actions from '../redux/actionCreators/Actions';
import BasketControl from './BasketControl';
import Receipt from './Receipt';
import PriceBox from './PriceBox';

const styles = theme => ({
  root: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'row',
    margin: '0 auto',
    backgroundColor: 'rgb(237,240,249)',
    flexWrap: 'wrap',
    overflow: 'scroll',
    height: '100%',
  },
  basket: {
    width: '40%',
    justifyContent: 'center',
    flexDirection: 'column',
    padding: '16px',
    flex: '0 0 auto',
    flexGrow: 1,
  },
});

class SelfCheckout extends React.Component {
  async componentDidMount() {
    const { getProducts, getScannedItems, getPromotions } = this.props;
    await getProducts();
    await getPromotions();
    await getScannedItems();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.root}>
        {/**
          * render the inventory component
          */}
        <BasketControl />
        <div className={classes.basket}>
          {/**
            * render the Basket component
            */}
          <Receipt />
          {/**
            * render the price
            */}
          <PriceBox />
        </div>
      </Paper>
    );
  }
}

SelfCheckout.propTypes = {
  getScannedItems: PropTypes.func.isRequired,
  getProducts: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
};

const mapDispatchToProps = dispatch => ({
  getProducts: () => dispatch(Actions.getProducts()),
  getScannedItems: () => dispatch(Actions.getScannedItems()),
  getPromotions: () => dispatch(Actions.getPromotions()),
});
export default connect(null, mapDispatchToProps)(withStyles(styles)(SelfCheckout));
