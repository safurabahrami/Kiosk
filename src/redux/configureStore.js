import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import RootReducer from './reducers';

const composeEnhancers = composeWithDevTools({
    // options like actionSanitizer, stateSanitizer
});

export const store = createStore(RootReducer, /* preloadedState, */ composeEnhancers(
    applyMiddleware(thunk)
    // other store enhancers if any
));