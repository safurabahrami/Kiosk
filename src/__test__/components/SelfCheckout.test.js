import React from 'react';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

import SelfCheckout from '../../components/SelfCheckout';

const mockStore = configureStore();
const store = mockStore({});

describe('renders without crashing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SelfCheckout store={store} />);
  });

  it('+++ render the component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('+++ match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
