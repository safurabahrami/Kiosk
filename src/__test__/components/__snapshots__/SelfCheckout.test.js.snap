// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing +++ match snapshot 1`] = `
<WithStyles(SelfCheckout)
  getProducts={[Function]}
  getPromotions={[Function]}
  getScannedItems={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
